File: admin-lite-mobile/scripts/generate-icons.js
#!/usr/bin/env node
// Regenerate app icons from the indigo "AL" glyph.
// Reads the 512x512 source from the web admin-lite's public/ folder.

const path = require('path');
const sharp = require('/tmp/node_modules/sharp');

const OUT = path.resolve(__dirname, '..', 'assets');

const glyphSvg = (size, pad = 0) => `
<svg xmlns="http://www.w3.org/2000/svg" width="${size}" height="${size}" viewBox="0 0 ${size} ${size}">
  ${pad > 0 ? `<rect width="${size}" height="${size}" fill="#4F46E5"/>` : ''}
  <text x="50%" y="50%" dominant-baseline="central" text-anchor="middle"
    font-family="system-ui, -apple-system, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif"
    font-weight="800" font-size="${Math.floor(size * 0.43)}" fill="#ffffff">AL</text>
</svg>`;

const adaptiveSvg = () => `
<svg xmlns="http://www.w3.org/2000/svg" width="1024" height="1024" viewBox="0 0 1024 1024">
  <text x="50%" y="50%" dominant-baseline="central" text-anchor="middle"
    font-family="system-ui, -apple-system, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif"
    font-weight="800" font-size="360" fill="#ffffff">AL</text>
</svg>`;

const splashSvg = () => `
<svg xmlns="http://www.w3.org/2000/svg" width="432" height="432" viewBox="0 0 432 432">
  <rect width="432" height="432" rx="96" fill="#ffffff" fill-opacity="0"/>
  <text x="50%" y="50%" dominant-baseline="central" text-anchor="middle"
    font-family="system-ui, -apple-system, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif"
    font-weight="800" font-size="220" fill="#ffffff">AL</text>
</svg>`;

(async () => {
  // Main app icon: 1024x1024 rounded-rect indigo with AL
  const iconSvg = `
<svg xmlns="http://www.w3.org/2000/svg" width="1024" height="1024" viewBox="0 0 1024 1024">
  <rect width="1024" height="1024" rx="224" fill="#4F46E5"/>
  <text x="50%" y="50%" dominant-baseline="central" text-anchor="middle"
    font-family="system-ui, -apple-system, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif"
    font-weight="800" font-size="440" fill="#ffffff">AL</text>
</svg>`;
  await sharp(Buffer.from(iconSvg)).png().toFile(path.join(OUT, 'icon.png'));
  // Adaptive icon foreground: AL glyph sized to fit in the 66% safe circle
  await sharp(Buffer.from(adaptiveSvg())).png().toFile(path.join(OUT, 'adaptive-icon.png'));
  // Splash image: small glyph, transparent background (indigo supplied via backgroundColor)
  await sharp(Buffer.from(splashSvg())).png().toFile(path.join(OUT, 'splash-icon.png'));
  // Favicon for web (harmless even though we don't export web)
  await sharp(Buffer.from(iconSvg)).resize(48, 48).png().toFile(path.join(OUT, 'favicon.png'));
  console.log('Icons generated at', OUT);
})().catch((e) => {
  console.error(e);
  process.exit(1);
});

File: admin-lite-mobile/scripts/generate-icons.py
#!/usr/bin/env python3
"""Regenerate the Expo app icons from the black "AL" wordmark.

Writes into ./assets:
  - icon.png          1024  rounded black square + white "AL"  (iOS / base)
  - adaptive-icon.png 1024  white "AL" on transparent (Android foreground;
                            background colour supplied via app.json)
  - splash-icon.png    432  white "AL" on transparent (splash bg via app.json)
  - favicon.png         48  rounded black square + white "AL"

Self-contained: uses Pillow (PIL), which ships in the toolchain, plus system
Arial Bold. Run:  python3 scripts/generate-icons.py
"""

import os

from PIL import Image, ImageDraw, ImageFont

ASSETS = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "assets"))

TEXT = "AL"
BG = (0, 0, 0, 255)            # black
FG = (255, 255, 255, 255)      # white
FONT_PATH = "/System/Library/Fonts/Supplemental/Arial Bold.ttf"
SS = 4  # supersample


def _fit_font(draw, text, target_w):
    lo, hi = 4, 8000
    while lo < hi:
        mid = (lo + hi + 1) // 2
        f = ImageFont.truetype(FONT_PATH, mid)
        l, t, r, b = draw.textbbox((0, 0), text, font=f)
        if (r - l) <= target_w:
            lo = mid
        else:
            hi = mid - 1
    return ImageFont.truetype(FONT_PATH, lo)


def render(size, *, width_ratio, corner_ratio=0.0, transparent_bg=False):
    W = size * SS
    img = Image.new("RGBA", (W, W), (0, 0, 0, 0))
    d = ImageDraw.Draw(img)
    if not transparent_bg:
        if corner_ratio > 0:
            d.rounded_rectangle([0, 0, W - 1, W - 1], radius=int(W * corner_ratio), fill=BG)
        else:
            d.rectangle([0, 0, W, W], fill=BG)
    font = _fit_font(d, TEXT, target_w=width_ratio * W)
    l, t, r, b = d.textbbox((0, 0), TEXT, font=font)
    x = (W - (r - l)) / 2 - l
    y = (W - (b - t)) / 2 - t
    d.text((x, y), TEXT, font=font, fill=FG)
    return img.resize((size, size), Image.LANCZOS)


def save(img, name):
    path = os.path.join(ASSETS, name)
    img.save(path)
    print("  wrote assets/" + name)


if __name__ == "__main__":
    os.makedirs(ASSETS, exist_ok=True)
    # Main icon: rounded black square + white AL
    save(render(1024, width_ratio=0.56, corner_ratio=0.22), "icon.png")
    # Adaptive foreground: white AL within the safe zone, transparent bg
    save(render(1024, width_ratio=0.42, transparent_bg=True), "adaptive-icon.png")
    # Splash glyph: white AL on transparent (bg colour from app.json)
    save(render(432, width_ratio=0.50, transparent_bg=True), "splash-icon.png")
    # Web favicon
    save(render(48, width_ratio=0.56, corner_ratio=0.22), "favicon.png")
    print("Done.")
